presubmit_tests/single-instance-on-gcp.sh

```diff
@@ -0,0 +1,153 @@
+sed -i "s|@instance_name@|$instance_name|g" "${tfvars_file}"
+
+echo "Applying Infra Manager deployment: ${deployment_id}"
+gcloud infra-manager deployments apply "${deployment_id}" \
+  --service-account="projects/${project}/serviceAccounts/infra-manager-deployer@${project}.iam.gserviceaccount.com" \
+  --local-source="./terraform" \
+  --inputs-file="${tfvars_file}" || exit 1
+
+echo "List resources for a deployment revision:"
+gcloud infra-manager resources list \
+--deployment="${deployment_name}" \
+--location="${location}" \
+--revision=r-0 \
+--format=json
+
+# Extract the id of the control node resource
+# The format is: projects/<project>/zones/<zone>/instances/control-node-<random-suffix>
+control_node_resource_id="$(gcloud infra-manager resources list \
+--deployment="${deployment_name}" \
+--location="${location}" \
+--revision=r-0 \
+--format=json | jq -r '.[] | select(.terraformInfo.address == "google_compute_instance.control_node") | .terraformInfo.id')"
+
+if [[ -z "${control_node_resource_id}" ]]; then
+  echo "Could not retrieve control node's resource ID."
+  exit 1
+fi
+echo "Control node resource ID: ${control_node_resource_id}"
+
+control_node_instance_zone="$(echo "${control_node_resource_id}" | cut -d'/' -f4)"
+if [[ -z "${control_node_instance_zone}" ]]; then
+  echo "Could not extract control node's zone."
+  exit 1
+fi
+echo "Control node zone: ${control_node_instance_zone}"
+
+control_node_instance_name="$(echo "${control_node_resource_id}" | cut -d'/' -f6)"
+if [[ -z "${control_node_instance_name}" ]]; then
+  echo "Could not extract control node's name."
+  exit 1
+fi
+echo "Control node name: ${control_node_instance_name}"
+
+# Get the instance ID from the instance name
+control_node_instance_id="$(gcloud compute instances describe "${control_node_instance_name}" \
+  --zone="${control_node_zone}" \
+  --format="value(id)")"
+if [[ -z "${control_node_instance_id}" ]]; then
+  echo "Could not get control node's instance ID."
+  exit 1
+fi
+echo "Control node instance ID: ${control_node_instance_id}"
+
+# Stream logs from the startup script execution to stdout in the background
+gcloud beta logging tail \
+"resource.type=gce_instance AND \
+resource.labels.instance_id=${control_node_instance_id} \
+AND log_name=projects/${project}/logs/google_metadata_script_runner" \
+--format='value(timestamp, json_payload.message)' &
+
+tail_pid=$!
+
+sleep_seconds=60
+timeout_seconds=7200
+timeout_hours="$((timeout_seconds / 3600))"
+timeout_minutes="$(((timeout_seconds % 3600) / 60))"
+start_time="$(date +%s)"
+
+echo "Waiting up to ${timeout_hours} hours and ${timeout_minutes} minutes for control node's startup script to complete..."
+
+while true; do
+  current_time="$(date +%s)"
+  elapsed="$((current_time - start_time))"
+
+  if [[ "${elapsed}" -ge "${timeout_seconds}" ]]; then
+    echo "Timeout reached after ${timeout_hours} hours and ${timeout_minutes} minutes. No completion log found."
+    exit 1
```

Comment:
Should we call for an explicit cleanup here?  The deployment is likely still around.